### PR TITLE
Apron, Cloak, Mantle rework, additions, restoration

### PR DESCRIPTION
**Pull request by Pebblesthe2nd**
## About The Pull Request

Offshoot of the big armor/suits rework. Have to partition it or go insane.

What is this?
Basically expands the clothing layer options by adding the categories mantles and aprons to the accessory slot.
Both have a mainly aesthetic function, with some minor practical stuff like limited storage, small resist or climate bonuses.

Currently there is clothing called cloaks, mantles, capes that either go on suit or on accessory slot, either have togglable armored hood or not, either have thick armor or none, with zero visual or name difference.
This changes that issue, and adds both functionality and customization options.
If it has a hood and armor values - it is a cloak, goes on suit slot.
If it has no hood (nothing that cover the face/gives armor specifically) and no armor, its a mantle.

* Two new data files for these clothing added, two new icon files to contain the sprites as well.
* All clothing in current assets that fits into the categories and has acceptable sprites copied there and made to adhere to the templates, and all instances of cloak replaced with mantle for those items.
* Added a number of various storage options with special pockets for the differing types of clothing created.
*  Current clothing types : Mantles, Ponchos, Aprons, Pelts (mantle style ones, not the head variants yet.)
* New art made for : Labor apron, Bartender apron, Housewife apron, Surgical apron, Poncho
* New clothing created + art : Mountain bear pelt, Fur pelt (longhorner), Ragged pelt (art the old unused lizard mantle so not mine), Gecko mantle.

Follow up for this PR will be some crafting recipes, replacing the items needed, and removing now obsolete items and art.

### CODE REVIEW

File: code/datums/components/storage/concrete/pockets.dm [hunk 1]
```diff
@@ -35,6 +35,94 @@
...
+		/obj/item/retractor,
+		/obj/item/cautery,
+		/obj/item/hemostat,
+		/obj/item/clothing/neck/stethoscope,
+		/obj/item/clothing/glasses,
+		/obj/item/storage/bag/chemistry,
+		/obj/item/storage/bag/bio,
+		/obj/item/reagent_containers/blood,
+		/obj/item/reagent_containers/chem_pack,
+		))
```
Comment (Rohesie) [inline code/datums/components/storage/concrete/pockets.dm:124]: Please turn these into global lists, not to produce a unique typecache each time.

File: code/modules/clothing/neck/apron.dm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	desc = "(I) Your classic, non-racist poncho. This one is green."
+	icon_state = "greenponcho"
+	item_state = "greenponcho"
+
+/obj/item/clothing/suit/poncho/red
+	name = "red poncho"
+	desc = "(I) Your classic, non-racist poncho. This one is red."
+	icon_state = "redponcho"
+	item_state = "redponcho"
+*/
```
Comment (Rohesie) [inline code/modules/clothing/neck/apron.dm:80]:
```suggestion

```
Remove commented-out code, and remove the sprites if unused.

File: code/modules/clothing/neck/mantle.dm [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	body_parts_covered = CHEST|ARMS
+	max_heat_protection_temperature = ARMOR_MAX_TEMP_PROTECT
+	min_cold_protection_temperature = ARMOR_MIN_TEMP_PROTECT
+	heat_protection = null
+	cold_protection = null
+	permeability_coefficient = 0.9
+
+/obj/item/clothing/neck/cloak/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is strangling [user.p_them()]self with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return(OXYLOSS)
```
Comment (Rohesie) [inline code/modules/clothing/neck/mantle.dm:18]:
```suggestion
```
I think we can live without this memery.

File: code/modules/clothing/neck/mantle.dm [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	desc = "(I) Your classic, non-racist poncho. This one is green."
+	icon_state = "greenponcho"
+	item_state = "greenponcho"
+
+/obj/item/clothing/suit/poncho/red
+	name = "red poncho"
+	desc = "(I) Your classic, non-racist poncho. This one is red."
+	icon_state = "redponcho"
+	item_state = "redponcho"
+*/
```
Comment (Rohesie) [inline code/modules/clothing/neck/mantle.dm:135]:
```suggestion
```

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: code/modules/clothing/neck/mantle.dm (hunk 4 above)
Comment (Pebblesthe2nd) [reply code/modules/clothing/neck/mantle.dm:135]: Need it to remain for now as a reminder for myself where to start the cleanup PR, as discussed.

**Pull request merged**